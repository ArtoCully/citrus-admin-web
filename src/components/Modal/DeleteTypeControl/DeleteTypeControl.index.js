// @flow
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { editDeviceTypes } from 'src/redux/modules/device-type'
import DeleteTypeControl from './DeleteTypeControl'

type StateProps = {}

function mapStateToProps (
  state: GlobalReducerState,
): StateProps {

  return {}

}

type DispatchProps = {
  editDeviceTypes: Function,
}

function mapDispatchToProps (dispatch: any): DispatchProps {

  return {
    editDeviceTypes: bindActionCreators(editDeviceTypes, dispatch),
  }

}

export default connect(mapStateToProps, mapDispatchToProps)(DeleteTypeControl)
export type ReduxProps = StateProps
