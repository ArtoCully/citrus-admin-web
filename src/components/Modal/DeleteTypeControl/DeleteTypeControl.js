// @flow
import React, { PureComponent } from 'react'
import LinkReplace from 'src/components/shared/LinkReplace/LinkReplace'
import css from './DeleteTypeControl.style.css'
import type { ReduxProps } from './DeleteTypeControl.index'


type Props = ReduxProps & {
  location: Object,
}

type State = {
  location: {
    typeControl: Object
  },
}

class DeleteTypeControl extends PureComponent {

  state: State

  delete = (index: Number) => {

    const deviceType = {
      ...this.props.location.state.deviceType,
      controls: [
        ...this.props.location.state.deviceType.controls,
      ],
    }
    deviceType.controls.splice(index, 1)
    this.props.editDeviceTypes(deviceType)

  }

  props: Props

  render (): ?React$Element<any> {

    const {
      location: {
        state: {
          typeControl = {},
        },
        pathname,
      },
    } = this.props

    return (
      <div className={css.deleteTypeControl}>
        <div className={css.header}>
          <div className={css.title}>DELETE TYPE CONTROL</div>
          <div className="pt4 pb4">
            Are you sure you want to delete {typeControl.name}?
          </div>
        </div>
        <LinkReplace to={pathname}>
          <button className={css.btnCancel}>Cancel</button>
        </LinkReplace>
        <LinkReplace to={pathname}>
          <button
            className={`${css.btnDelete} ml2`}
            onClick={() => {

              this.delete(typeControl.index)

            }}
          >
            Delete
          </button>
        </LinkReplace>
      </div>
    )

  }

}

export default DeleteTypeControl
