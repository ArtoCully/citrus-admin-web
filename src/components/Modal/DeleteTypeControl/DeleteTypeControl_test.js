// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import DeleteTypeControl from './DeleteTypeControl'

const mockProps = {
  location: {
    pathname: '/',
    state: {
      deviceType: {},
      typeControl: {},
    },
  },
  deviceType: {},
}


it('<DeleteTypeControl> renders the DeleteTypeControl content', () => {

  const wrapper = shallow(
    <DeleteTypeControl
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
