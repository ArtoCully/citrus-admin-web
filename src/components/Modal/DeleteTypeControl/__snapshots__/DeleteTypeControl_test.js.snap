// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteTypeControl> renders the DeleteTypeControl content 1`] = `
<div
  className="deleteTypeControl"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      DELETE TYPE CONTROL
    </div>
    <div
      className="pt4 pb4"
    >
      Are you sure you want to delete 
      ?
    </div>
  </div>
  <LinkReplace
    to="/"
  >
    <button
      className="btnCancel"
    >
      Cancel
    </button>
  </LinkReplace>
  <LinkReplace
    to="/"
  >
    <button
      className="btnDelete ml2"
      onClick={[Function]}
    >
      Delete
    </button>
  </LinkReplace>
</div>
`;
